### PR TITLE
updates to BPs

### PR DESCRIPTION
**Pull request by bmoore-msft**
### Description of the change

Updates to meet BPs, simplified nesting

### CODE REVIEW

File: 301-multi-vmss-linux/azuredeploy.json [hunk 1]
```diff
@@ -83,21 +87,18 @@
       "properties": {
```
Comment (philon-msft) [inline 301-multi-vmss-linux/azuredeploy.json]: Latest schema apiVersion for Microsoft.Resources/deployments is 2016-02-01

File: 301-multi-vmss-linux/azuredeploy.parameters.json [hunk 2]
```diff
@@ -2,17 +2,14 @@
...
-    },
     "numberOfVMSS": {
       "value": 5
     },
     "instanceCountPerVMSS": {
       "value": 5
     },
     "adminUsername": {
-      "value": "multi"
+      "value": "ghuser"
```
Comment (philon-msft) [inline 301-multi-vmss-linux/azuredeploy.parameters.json]: Could be GEN-USER

File: 301-multi-vmss-linux/nestedtemplates/base.json [hunk 3]
```diff
@@ -27,38 +24,39 @@
   "variables": {
     "imagePublisher": "Canonical",
     "imageOffer": "UbuntuServer",
-    "imageVersion": "15.10",
+    "osVersion": "16.04.0-LTS",
```
Comment (philon-msft) [inline 301-multi-vmss-linux/nestedtemplates/base.json]: Consider eliminating variables like these that hold values that the schemas could be suggesting allowedValues for

File: 301-multi-vmss-linux/nestedtemplates/base.json [hunk 4]
```diff
@@ -80,11 +78,11 @@
           "storageProfile": {
             "osDisk": {
               "vhdContainers": [
-                "[concat('https://', variables('uniqueStringArray')[0], parameters('newStorageAccountPrefix'), '.blob.core.windows.net/', variables('vhdContainerName'))]",
-                "[concat('https://', variables('uniqueStringArray')[1], parameters('newStorageAccountPrefix'), '.blob.core.windows.net/', variables('vhdContainerName'))]",
-                "[concat('https://', variables('uniqueStringArray')[2], parameters('newStorageAccountPrefix'), '.blob.core.windows.net/', variables('vhdContainerName'))]",
-                "[concat('https://', variables('uniqueStringArray')[3], parameters('newStorageAccountPrefix'), '.blob.core.windows.net/', variables('vhdContainerName'))]",
-                "[concat('https://', variables('uniqueStringArray')[4], parameters('newStorageAccountPrefix'), '.blob.core.windows.net/', variables('vhdContainerName'))]"
+                "[concat(reference(resourceId('Microsoft.Storage/storageAccounts/', variables('uniqueStringArray')[0]), '2016-01-01').primaryEndpoints.blob, 'vhds')]",
```
Comment (philon-msft) [inline 301-multi-vmss-linux/nestedtemplates/base.json]: apiVersion isn't necessary in the reference() call if the resource is provisioned in the same template. The 'Microsoft.Storage/storageAccounts/' prefix should be necessary either.

File: 301-multi-vmss-linux/nestedtemplates/base.json [hunk 5]
```diff
@@ -27,38 +24,39 @@
...
+      "[concat(uniqueString(resourceGroup().id, '0'), '0')]",
+      "[concat(uniqueString(resourceGroup().id, '1'), '1')]",
+      "[concat(uniqueString(resourceGroup().id, '2'), '2')]",
+      "[concat(uniqueString(resourceGroup().id, '3'), '3')]",
+      "[concat(uniqueString(resourceGroup().id, '4'), '4')]"
     ],
-    "saCount": "[length(variables('uniqueStringArray'))]",
-    "computeApiVersion": "2016-03-30",
-    "storageApiVersion": "2015-06-15"
+    "saCount": "[length(variables('uniqueStringArray'))]"
```
Comment (philon-msft) [inline 301-multi-vmss-linux/nestedtemplates/base.json]: No need to have an saCount variable - you could just inline the call to [length(variables('uniqueStringArray'))]

File: 301-multi-vmss-linux/azuredeploy.parameters.json (hunk 2 above)
Comment (bmoore-msft) [inline 301-multi-vmss-linux/azuredeploy.parameters.json]: done

File: 301-multi-vmss-linux/nestedtemplates/base.json (hunk 3 above)
Comment (bmoore-msft) [inline 301-multi-vmss-linux/nestedtemplates/base.json]: done

File: 301-multi-vmss-linux/nestedtemplates/base.json (hunk 5 above)
Comment (bmoore-msft) [inline 301-multi-vmss-linux/nestedtemplates/base.json]: done

File: 301-multi-vmss-linux/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [inline 301-multi-vmss-linux/azuredeploy.json]: done

File: 301-multi-vmss-linux/nestedtemplates/base.json (hunk 4 above)
Comment (bmoore-msft) [inline 301-multi-vmss-linux/nestedtemplates/base.json]: done

**Pull request merged**